src/TerrainRenderer.cpp

```diff
@@ -349,18 +347,15 @@ void TerrainRenderer::UpdateTriangleTerrain(const MapPoint pt, bool updateVBO)
         int w = texRect.right - texRect.left;
         int h = texRect.bottom - texRect.top;
         RTTR_Assert(w > 0 && h > 0);
-        unsigned texW = helpers::roundToNextPowerOfTwo(w);
-        unsigned texH = helpers::roundToNextPowerOfTwo(h);
+        Point<float> texSize(VIDEODRIVER.calcPreferredTextureSize(Extent(w, h)));
 
-        float texScaleW = 1.f / texW;
-        float texScaleH = 1.f / texH;
         // Tip of the triangle is in the middle in x
-        texCoord[1].x = (w + 1) / 2.f * texScaleW;
+        texCoord[1].x = (w + 1) / 2.f / texSize.x;

```

Comment:
use multiplications these are much faster

Reply:
Not if you do the division before. I now leave it to the compiler to figure that out. Most likely the asm will be the same but the code is more clear